File: Week 07/id_372/number_of_1_bits/number_of_1_bits.go
package number_of_1_bits

func hammingWeight(num uint32) int {
	count := 0
	for ; num > 0; num >>= 1 {
		if num&1 == 1 {
			count++
		}
	}
	return count
}

func hammingWeight1(num uint32) int {
	count := 0
	for ; num > 0; num &= num - 1 {
		count++
	}

	return count
}

File: Week 07/id_277/number-of-1-bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        return bin(n).count('1')

class Solution:
    def hammingWeight(self, n: int) -> int:
        n = bin(n)
        count = 0
        for i in n:
            if i == '1':
                count += 1
        return count